### PR TITLE
chore: add dco action

### PR DESCRIPTION
**Pull request by mattrosno**
Adds GitHub Action to enforce Developer Certificate of Origin (DCO) signatures. Read more here: https://github.com/carbon-design-system/carbon-dco

Configuration:

1. I added the `carbon-bot` PERSONAL_ACCESS_TOKEN to this repository as a secret
2. I made sure to exclude dependabot and kodiakhq

### CODE REVIEW

File: .github/workflows/dco.yml
Comment (tw15egan) [inline .github/workflows/dco.yml:22]: Will Vercel also trigger this when it comments the deploy previews?

What about PR's by another action like https://github.com/carbon-design-system/carbon-website/pull/2177

**Review by tw15egan — APPROVED** [review]
A small comment, but verified it works on the test repo 👍🏻

File: .github/workflows/dco.yml
Comment (mattrosno) [reply .github/workflows/dco.yml:22]: `kodiakhq` bot makes the commits for the image compression PRs, so I think we're good there. E.g. https://github.com/carbon-design-system/carbon-website/pull/2177/commits/2b48d3b5a44e16d169c49ce0135ce4757594398e

I'm unsure if this check runs on every comment like Vercel deploy previews. If the action is slowing down our review, we can always disable it and look for alternative solutions.

Comment (tw15egan) [reply .github/workflows/dco.yml:22]: Yeah, just something I was thinking of, definitely not a blocker. Thanks for adding this in!

**Review by tay1orjones — APPROVED** [review]

**Review by joshblack — APPROVED** [review]

**Comment by mattrosno** [comment]
I'm going to update this to use a newer version of the GitHub Action, then we can merge it in.

**Comment by mattrosno** [comment]
Testing the check. It takes a minute or so before the check runs.
image

Signing the DCO.
image

It took a few minutes for the DCO to automatically run again after signing (commenting).
image

The check passes.
image

`carbon-bot` added this commit in the `carbon-dco` repo to save the signature. https://github.com/carbon-design-system/carbon-dco/commit/f39e6cc5da7cadb3eb5fc6273e46cf3318925c44

**Comment by mattrosno** [comment]
@tw15egan let's wait on this issue that kodiak's primary maintainer just opened. https://github.com/chdsbd/kodiak/issues/641

I can also do a test with kodiak to ensure if this is just an issue with this specific PR because it adds the GH action that references the repo secret (meaning we can just merge this PR without the check), or if it'll be an ongoing issue with future PRs after this has been merged.

**Pull request merged**
